packages/backend-app-api/src/services/implementations/cacheService.ts

```diff
@@ -18,20 +18,21 @@ import { CacheManager } from '@backstage/backend-common';
 import {
   configServiceRef,
   createServiceFactory,
+  pluginMetadataServiceRef,
   cacheServiceRef,
 } from '@backstage/backend-plugin-api';
 
 /** @public */
 export const cacheFactory = createServiceFactory({
   service: cacheServiceRef,
   deps: {
-    configFactory: configServiceRef,
+    config: configServiceRef,
+    plugin: pluginMetadataServiceRef,
   },
-  factory: async ({ configFactory }) => {
-    const config = await configFactory('root');
+  async factory({ config }) {
     const cacheManager = CacheManager.fromConfig(config);
-    return async (pluginId: string) => {
-      return cacheManager.forPlugin(pluginId);
+    return async ({ plugin }) => {
+      return cacheManager.forPlugin(plugin.getId());
```

Comment:
Syntax sugar: Maybe could deconstruct also the id property, as for example:
```javascript
    return async ({ plugin: { id: pluginId } }) => {
      return cacheManager.forPlugin(pluginId);
```

Reply:
The `PluginMetadataService` as it is atm is unlikely to be final, so there's definitely space to tweak things. I think making the ID a simple prop might make sense. In general I think it's best to stick to methods in interfaces though, just in case the implementation needs some form of side-effects.